pkg/storage/node_liveness.go

```diff
@@ -67,6 +69,32 @@ var (
 	}
 )
 
+// livenessSlice to sort Liveness records and compute a hash of the contents.
+type livenessSlice []Liveness
+
+// livenessSlice implements sort.Interface.
+var _ sort.Interface = livenessSlice{}
+
+func (ls livenessSlice) Len() int { return len(ls) }
+
+func (ls livenessSlice) Less(i, j int) bool { return ls[i].NodeID < ls[j].NodeID }
+
+func (ls livenessSlice) Swap(i, j int) { ls[i], ls[j] = ls[j], ls[i] }
+
+// Hash returns a SHA1 hash of the contents.
+func (ls livenessSlice) Hash() []byte {
+	sort.Sort(ls)
```

Comment:
Feels sort of strange that `Hash` has the side-effect of `Sorting`. Perhaps this should be inlined in `NodeLiveness.Hash`.